### PR TITLE
Ft fast evaluate derivatives

### PR DESCRIPTION
**Pull request by jzwar**
# Addressed issues
- #4 

# Description
Evaluate derivatives directly at a certain point without the need to build the derivative spline first. Similar to what SplineLib is doing.

## New Feature Showcase 
```
// Requested derivatives
const std::array<std::size_t, 3> derivatives{1,0,1};
// Given a 3D->3D spline of degrees [3,2,4]
PointType derivative = spline.EvaluateDerivative(evaluation_point, derivatives);
// Which is equivalent to
PointType derivative = spline
    .DerivativeWRTParametricDimension(0)
    .DerivativeWRTParametricDimension(2)
    .Evaluate(evaluation_point);
```

# Check list
### Test documentation
* [x]  Build Debug
* [x]  Build Release Mode
* [x]  Executed Tests
* [x]  Build Tests
* [x]  Executed Examples
* [x] Added a new unit-test for feature

### Documentation of changes and theory
* [x]  I have updated the relevant documentations, as far as necessary.

### Style guide compliance
* [x]  Edited `C++` files formatted with clang-format (`10.x.x`+)

### CODE REVIEW

**Comment by jzwar** [comment]
No longer WIP

**Review by j042 — APPROVED** [review]
Nice feature! I guess think will be soon available in splinepy?

File: src/bezier_spline.inc [hunk 1]
```diff
@@ -351,6 +331,48 @@ constexpr PhysicalPointType BezierSpline<
...
+template <std::size_t parametric_dimension, typename PhysicalPointType,
+          typename ScalarType>
+constexpr PhysicalPointType
+BezierSpline<parametric_dimension, PhysicalPointType, ScalarType>::
+    EvaluateDerivative(
+        const PointTypeParametric_& par_coords,
+        const std::array<std::size_t, parametric_dimension>& nth_derivs) const {
+  /*
+   * Here we avoid Point operations and try to precompute as many values as
+   * possible, this greatly increases the complexity of the necessary code.
```
[the remark this answers is not in the document]
Comment (jzwar) [reply src/bezier_spline.inc:362]: ups :D well

**Review by j042 — APPROVED** [review]
Thanks, Awesome!

**Pull request merged**